flink-python/pyflink/datastream/async_data_stream.py

```diff
@@ -47,14 +48,39 @@ def unordered_wait(
         :param output_type: The output data type.
         :return: The transformed DataStream.
         """
-        AsyncDataStream._validate(data_stream, async_function)
+        return AsyncDataStream.unordered_wait_with_retry(
+            data_stream, async_function, timeout, async_retry_strategies.NO_RETRY_STRATEGY,
+            capacity, output_type)
+
+    @staticmethod
+    def unordered_wait_with_retry(
```

Comment:
why do we prefer introducing `unordered_wait_with_retry` instead of adding a new parameter(AsyncRetryStrategy, default is None) to `unordered_wait`

Reply:
I mean this https://github.com/apache/flink/blob/5b61b1cda997fa28098948297e2d5e91d7517f48/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/AsyncDataStream.java#L217